packages/ag-charts-enterprise/src/features/shared-toolbar/sharedToolbar.ts

```diff
@@ -1,20 +1,168 @@
 import { _ModuleSupport } from 'ag-charts-community';
 
-import { SharedToolbarWidget } from './sharedToolbarWidget';
+import type { SharedToolbarSection } from './sharedToolbarTypes';
+
+export interface SharedToolbarWithSection<
+    ButtonOptions extends _ModuleSupport.ToolbarButtonOptions = _ModuleSupport.ToolbarButtonOptions,
+> extends Pick<
+        _ModuleSupport.Toolbar<ButtonOptions>,
+        | 'destroy'
+        | 'addListener'
+        | 'removeListener'
+        | 'setHidden'
+        | 'addToolbarListener'
+        | 'updateButtons'
+        | 'updateButtonByIndex'
+        | 'toggleActiveButtonByIndex'
+        | 'toggleButtonEnabledByIndex'
+        | 'clearActiveButton'
+    > {
+    layout: (layoutBox: _ModuleSupport.BBox) => void;
+}
 
 export class SharedToolbar extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
-    private sharedToolbar?: SharedToolbarWidget<_ModuleSupport.ToolbarButtonOptions>;
+    static readonly SECTION_ORDER: Array<SharedToolbarSection> = ['chartToolbar', 'annotations'];
+
+    private sharedToolbar?: _ModuleSupport.Toolbar<_ModuleSupport.ToolbarButtonOptions>;
+    private readonly activeSections: Set<SharedToolbarSection> = new Set();
+    private readonly sectionButtons: Record<SharedToolbarSection, Array<_ModuleSupport.ToolbarButtonOptions>> = {
+        annotations: [],
+        chartToolbar: [],
+    };
+    private firstLayoutSection?: SharedToolbarSection;
 
     constructor(private readonly ctx: _ModuleSupport.ModuleContext) {
         super();
     }
 
-    public getSharedToolbar<ButtonOptions extends _ModuleSupport.ToolbarButtonOptions>() {
+    public getSharedToolbar<ButtonOptions extends _ModuleSupport.ToolbarButtonOptions>(section: SharedToolbarSection) {
         if (!this.sharedToolbar) {
-            this.sharedToolbar = new SharedToolbarWidget(this.ctx.localeManager);
-            this.ctx.domManager.addChild('canvas-overlay', 'shared-toolbar', this.sharedToolbar.getElement());
+            this.createSharedToolbar();
+        }
+
+        return this.toolbarWithSection<ButtonOptions>(section);
+    }
+
+    private createSharedToolbar() {
+        this.sharedToolbar = new _ModuleSupport.Toolbar(this.ctx.localeManager, 'vertical');
+        this.sharedToolbar.addClass('ag-charts-shared-toolbar');
+
+        this.ctx.domManager.addChild('canvas-overlay', 'shared-toolbar', this.sharedToolbar.getElement());
+        this.destroyFns.push(() => {
+            this.sharedToolbar?.destroy();
+            this.sharedToolbar = undefined;
+            this.ctx.domManager.removeChild('canvas-overlay', 'shared-toolbar');
+        });
+    }
+
+    private toolbarWithSection<ButtonOptions extends _ModuleSupport.ToolbarButtonOptions>(
+        section: SharedToolbarSection
+    ): SharedToolbarWithSection<ButtonOptions> {
+        const sharedToolbar = this.sharedToolbar!;
+
+        const withSection = {
```

Comment:
This isn't entirely ideal, but the other option was duplicating tracking of active sections in both the `SharedToolbar` manager and the `SharedToolbarWidget`, along with the kinda meh duplication of functions with `section` prepended.

This essentially overrides the functions it needs to using the hidden section value from `.getSharedToolbar(section)`.